### PR TITLE
Improve routing documentation

### PR DESCRIPTION
**Pull request by ldez**
### What does this PR do?

- review routing sections and providers sections.
- review getting started.
- review entry point section.
- quick review of operation.
- add info on the static configuration migration.
- fix some typos

### Motivation

Have a better documentation.

### More

- [ ] ~~Added/updated tests~~
- [x] Added/updated documentation

### CODE REVIEW

File: docs/content/providers/docker.md [hunk 1]
```diff
@@ -539,94 +537,6 @@ providers:
...
-
-If a container is linked to several networks, be sure to set the proper network name (you can check this with `docker inspect <container_id>`), otherwise it will randomly pick one (depending on how docker is returning them).
-
-!!! warning
-    When deploying a stack from a compose file `stack`, the networks defined are prefixed with `stack`.
-
-#### `traefik.docker.lbswarm`
-
-Enables Swarm's inbuilt load balancer (only relevant in Swarm Mode).
+## Routing Configuration
```
Comment (dduportal) [inline docs/content/providers/docker.md]: What do you think moving this section before the "Provider Configuration"? The argument would be that this content is most frequently read and accessed than the details of the Docker/Swarm provider config.

File: docs/content/providers/file.md [hunk 2]
```diff
@@ -96,15 +96,12 @@ You can write these configuration elements:
             passHostHeader: false
     ```
 
-## Provider Configuration Options
+## Provider Configuration
 
-!!! tip "Browse the Reference"
-    If you're in a hurry, maybe you'd rather go through the [static](../reference/static-configuration/overview.md) and the [dynamic](../reference/dynamic-configuration/file.md) configuration references.
+If you're in a hurry, maybe you'd rather go through the [static configuration](../reference/static-configuration/overview.md) and the [dynamic configuration](../reference/dynamic-configuration/file.md) references.
```
Comment (dduportal) [inline docs/content/providers/file.md]:
```suggestion
If you're in a hurry, maybe you'd rather go through the [dynamic configuration for the File provider](../reference/dynamic-configuration/file.md) references.
```

File: docs/content/providers/kubernetes-ingress.md [hunk 3]
```diff
@@ -47,10 +47,9 @@ spec:
           servicePort: 80
 ```
 
-## Provider Configuration Options
+## Provider Configuration
 
-!!! tip "Browse the Reference"
-    If you're in a hurry, maybe you'd rather go through the [static](../reference/static-configuration/overview.md) configuration reference.
+If you're in a hurry, maybe you'd rather go through the [static configuration](../reference/static-configuration/overview.md) reference.
```
Comment (dduportal) [inline docs/content/providers/kubernetes-ingress.md]:
```suggestion
```

File: docs/content/routing/entrypoints.md [hunk 4]
```diff
@@ -51,199 +56,414 @@ They define the port which will receive the requests (whether HTTP or TCP).
...
     
     ```bash tab="CLI"
+    ## Static configuration
     --entryPoints.web.address=:80
-    --entryPoints.web.proxyProtocol.trustedIPs=127.0.0.1/32,192.168.1.7
+    --entryPoints.web.forwardedHeaders.trustedIPs=127.0.0.1/32,192.168.1.7
     ```
 
-    IPs in `trustedIPs` only will lead to remote client address replacement: Declare load-balancer IPs or CIDR range here.
+??? info "`forwardedHeaders.insecure`"
```
Comment (dduportal) [inline docs/content/routing/entrypoints.md]:
```suggestion
??? info "`forwardedHeaders.insecure`: Insecure Mode (Always Trusting Forwarded Headers)."
```

File: docs/content/routing/entrypoints.md [hunk 5]
```diff
@@ -51,199 +56,414 @@ They define the port which will receive the requests (whether HTTP or TCP).
...
     
-??? example "Insecure Mode -- Testing Environment Only"
-
-    In a test environments, you can configure Traefik to trust every incoming connection.
-    Doing so, every remote client address will be replaced (`trustedIPs` won't have any effect)
+    Insecure Mode (Always Trusting Forwarded Headers).
```
Comment (dduportal) [inline docs/content/routing/entrypoints.md]:
```suggestion
```

File: docs/content/routing/entrypoints.md [hunk 6]
```diff
@@ -51,199 +56,414 @@ They define the port which will receive the requests (whether HTTP or TCP).
...
+    ```
 
-If proxyprotocol header parsing is enabled for the entry point, this entry point can accept connections with or without proxyprotocol headers.
+### Forwarded Header
 
-If the proxyprotocol header is passed, then the version is determined automatically.
+You can configure Traefik to trust the forwarded headers information (`X-Forwarded-*`).
 
-??? example "Enabling Proxy Protocol with Trusted IPs" 
+??? info "`forwardedHeaders.trustedIPs`"
```
Comment (dduportal) [inline docs/content/routing/entrypoints.md]:
```suggestion
??? info "`forwardedHeaders.trustedIPs`: Trusting Forwarded Headers from specific IPs."
```

File: docs/content/routing/entrypoints.md [hunk 7]
```diff
@@ -51,199 +56,414 @@ They define the port which will receive the requests (whether HTTP or TCP).
...
+    
+    Trusting Forwarded Headers from specific IPs.
```
Comment (dduportal) [inline docs/content/routing/entrypoints.md]:
```suggestion
```

File: docs/content/routing/entrypoints.md [hunk 8]
```diff
@@ -51,199 +56,414 @@ They define the port which will receive the requests (whether HTTP or TCP).
...
+    ## Static configuration
+    --entryPoints.name.address=:8888
+    --entryPoints.name.transport.lifeCycle.graceTimeOut=42
+    ```
+
+### ProxyProtocol
+
+Traefik supports [ProxyProtocol](https://www.haproxy.org/download/2.0/doc/proxy-protocol.txt) version 1 and 2.
+
+If proxyprotocol header parsing is enabled for the entry point, this entry point can accept connections with or without proxyprotocol headers.
```
Comment (dduportal) [inline docs/content/routing/entrypoints.md]:
```suggestion
If ProxyProtocol header parsing is enabled for the entrypoint, this entrypoint can accept connections with or without ProxyProtocol headers.
```

File: docs/content/routing/entrypoints.md [hunk 9]
```diff
@@ -51,199 +56,414 @@ They define the port which will receive the requests (whether HTTP or TCP).
...
+
+If the proxyprotocol header is passed, then the version is determined automatically.
```
Comment (dduportal) [inline docs/content/routing/entrypoints.md]:
```suggestion
If the ProxyProtocol header is passed, then the version is determined automatically.
```

File: docs/content/routing/entrypoints.md [hunk 10]
```diff
@@ -51,199 +56,414 @@ They define the port which will receive the requests (whether HTTP or TCP).
...
+
+??? info "`proxyProtocol.trustedIPs`"
```
Comment (dduportal) [inline docs/content/routing/entrypoints.md]:
```suggestion
??? info "`proxyProtocol.trustedIPs`: Enabling Proxy Protocol with Trusted IPs." 
```

File: docs/content/routing/entrypoints.md [hunk 11]
```diff
@@ -51,199 +56,414 @@ They define the port which will receive the requests (whether HTTP or TCP).
...
+
+Traefik supports [ProxyProtocol](https://www.haproxy.org/download/2.0/doc/proxy-protocol.txt) version 1 and 2.
+
+If proxyprotocol header parsing is enabled for the entry point, this entry point can accept connections with or without proxyprotocol headers.
+
+If the proxyprotocol header is passed, then the version is determined automatically.
+
+??? info "`proxyProtocol.trustedIPs`" 
+    
+    Enabling Proxy Protocol with Trusted IPs.
```
Comment (dduportal) [inline docs/content/routing/entrypoints.md]:
```suggestion
```

File: docs/content/routing/entrypoints.md [hunk 12]
```diff
@@ -51,199 +56,414 @@ They define the port which will receive the requests (whether HTTP or TCP).
...
     ```bash tab="CLI"
     --entryPoints.web.address=:80
-    --entryPoints.web.forwardedHeaders.trustedIPs=127.0.0.1/32,192.168.1.7
+    --entryPoints.web.proxyProtocol.trustedIPs=127.0.0.1/32,192.168.1.7
     ```
 
-??? example "Insecure Mode -- Always Trusting Forwarded Headers"
+    IPs in `trustedIPs` only will lead to remote client address replacement: Declare load-balancer IPs or CIDR range here.
+
+??? info "`proxyProtocol.insecure`"
```
Comment (dduportal) [inline docs/content/routing/entrypoints.md]:
```suggestion
??? info "`proxyProtocol.insecure`: Insecure Mode (Testing Environment Only)."
```

File: docs/content/routing/entrypoints.md [hunk 13]
```diff
@@ -51,199 +56,414 @@ They define the port which will receive the requests (whether HTTP or TCP).
...
+
+    Insecure Mode (Testing Environment Only).
```
Comment (dduportal) [inline docs/content/routing/entrypoints.md]:
```suggestion
```

File: docs/content/routing/entrypoints.md [hunk 14]
```diff
@@ -51,199 +56,414 @@ They define the port which will receive the requests (whether HTTP or TCP).
...
+    
+    ```bash tab="CLI"
+    ## Static configuration
+    --entryPoints.name.address=:8888
+    --entryPoints.name.transport.lifeCycle.graceTimeOut=42
+    ```
+
+### ProxyProtocol
+
+Traefik supports [ProxyProtocol](https://www.haproxy.org/download/2.0/doc/proxy-protocol.txt) version 1 and 2.
```
Comment (dduportal) [inline docs/content/routing/entrypoints.md]: I'm not sure if ProxyProtocol works with TCP routers/services (e.g. "can I enable ProxyProtocol for a MySQL server behind traefik?").

- If it is not, we have to add a "warning" admonition here.
- If it is supported, then this comment can be dismissed.

File: docs/content/getting-started/configuration-overview.md [hunk 15]
```diff
@@ -21,23 +21,25 @@ This configuration can change and is seamlessly hot-reloaded, without any reques
 
 ## The Dynamic Configuration 
 
-Traefik gets its _dynamic configuration_ from [providers](../providers/overview.md): whether an orchestrator, a service registry, or a plain old configuration file. Since this configuration is specific to your infrastructure choices, we invite you to refer to the [dedicated section of this documentation](../providers/overview.md).
+Traefik gets its _dynamic configuration_ from [providers](../providers/overview.md): whether an orchestrator, a service registry, or a plain old configuration file.
 
-!!! Note 
+Since this configuration is specific to your infrastructure choices, we invite you to refer to the [dedicated section of this documentation](../routing/overview.md).
+
+!!! info ""
```
[the remark this answers is not in the document]
Comment (ldez) [reply docs/content/getting-started/configuration-overview.md]: it's the expected behavior

File: docs/content/https/acme.md [hunk 16]
```diff
@@ -162,7 +162,7 @@ when using the `HTTP-01` challenge, `certificatesResolvers.sample.acme.httpChall
     --certificatesResolvers.sample.acme.httpChallenge.entryPoint=web
     ```
 
-!!! note
+!!! info ""
```
[the remark this answers is not in the document]
Comment (ldez) [reply docs/content/https/acme.md]: it's the expected behavior

File: docs/content/routing/entrypoints.md (hunk 4 above)
Comment (ldez) [reply docs/content/routing/entrypoints.md]: I prefer to be homogeneous with other element.
The name of the option can be long.

File: docs/content/routing/entrypoints.md (hunk 5 above)
Comment (ldez) [reply docs/content/routing/entrypoints.md]: I prefer to be homogeneous with other element.
The name of the option can be long.

File: docs/content/routing/entrypoints.md (hunk 6 above)
Comment (ldez) [reply docs/content/routing/entrypoints.md]: I prefer to be homogeneous with other element.
The name of the option can be long.

File: docs/content/routing/entrypoints.md (hunk 7 above)
Comment (ldez) [reply docs/content/routing/entrypoints.md]: I prefer to be homogeneous with other element.
The name of the option can be long.

File: docs/content/routing/entrypoints.md (hunk 10 above)
Comment (ldez) [reply docs/content/routing/entrypoints.md]: I prefer to be homogeneous with other element.
The name of the option can be long.

File: docs/content/routing/entrypoints.md (hunk 11 above)
Comment (ldez) [reply docs/content/routing/entrypoints.md]: I prefer to be homogeneous with other element.
The name of the option can be long.

File: docs/content/routing/entrypoints.md (hunk 12 above)
Comment (ldez) [reply docs/content/routing/entrypoints.md]: I prefer to be homogeneous with other element.
The name of the option can be long.

File: docs/content/routing/entrypoints.md (hunk 13 above)
Comment (ldez) [reply docs/content/routing/entrypoints.md]: I prefer to be homogeneous with other element.
The name of the option can be long.

File: docs/content/routing/entrypoints.md (hunk 14 above)
Comment (ldez) [reply docs/content/routing/entrypoints.md]: out of the scope of my PR for now.

**Pull request merged**